discretisedfield/field.py

```diff
@@ -108,12 +109,24 @@ class Field:
 
     """
 
-    def __init__(self, mesh, dim, value=0, norm=None):
+    def __init__(self, mesh, dim, value=0, norm=None, components=None):
         self.mesh = mesh
         self.dim = dim
         self.value = value
         self.norm = norm
 
+        self._components = None  # required for correct initialisation
+
+        if components is None:
+            if 2 <= dim <= 3:
+                components = ['x', 'y', 'z'][:dim]
+            elif dim > 3:
+                warnings.warn(f'Component labels must be specified for {dim=}'
+                              ' fields to get access to individual vector'
+                              ' components.')
+        if components is not None:
+            self.components = components
+
```

Comment:
This could be part of the property. No need to move it now, we can do that when we refactor.